cpp/util/etcd_test.cc

```diff
@@ -245,78 +256,106 @@ TEST_F(EtcdTest, TestGetAll) {
 }
 
 TEST_F(EtcdTest, TestCreate) {
-  EXPECT_CALL(client_, Generic(kEntryKey,
-                               AllOf(Contains(Pair(kValueParam, "123")),
-                                     Contains(Pair(kPrevExistParam, kFalse))),
-                               UrlFetcher::Verb::PUT, _, _))
-      .WillOnce(Invoke(
-          bind(&GenericReturn, Status::OK, MakeJson(kCreateJson), 1, _4, _5)));
+  EXPECT_CALL(
+      url_fetcher_,
+      Fetch(IsUrlFetchRequest(
+                UrlFetcher::Verb::PUT, URL(GetEtcdUrl(kEntryKey)),
+                ElementsAre(Pair(StrCaseEq("content-type"),
+                                 "application/x-www-form-urlencoded")),
+                "consistent=true&prevExist=false&quorum=true&value=123"),
+            _, _))
+      .WillOnce(
+          Invoke(bind(HandleFetch, Status::OK, 200,
+                      UrlFetcher::Headers{make_pair("x-etcd-index", "1")},
```

Comment:
I was implying that as I sort this out, and make the ```Response::etcd_index``` more rigorous, I'll tighten up the test (many of the tests aren't even checking the value of that field!).

I've seen actual etcd replies that had a lower ```X-Etcd-Index``` than the index in their JSON payload, so I guess we shouldn't be too strict on this... :wink: